### DISCUSSION TITLE
F# in a TypeScript Environment - Whats the state of Integration?

### DISCUSSION THREAD

**Discussion by D1no** [fable-compiler/Fable#2853] [Q&A] [answered]
Hello everyone. I'm interested in F# and would like to get some opinions on it's "friendliness" within TypeScript environments.

Concretely:
 - I'm evaluating the use of finite state machines that run both on the client and server to make state management and the tracking, reaction and handling of side effects easier.
 - F# is a natural application for state machines as I have seen in some examples.
 - However, for that I would not want to integrate .net within a (large) typescript code base and therefore wonder how well a F# via Fable / WebSharper would play along the lives of typescript developers?
 - In TypeScript land there is [xState](https://github.com/statelyai/xstate) for example. But TypeScript (like C#) is quite boilerplaty; something that F# could solve.
 - So considering to drop F# into a typescript codebase (with the hopes it could even be just a web worker / wasm file) it would be necessary to have good typed interfaces so TS devs would ideally not even realize they are interfacing with a state manager that is not typescript

I have read around here and saw that typescript is not really on the radar (?). It's not clear to me if that just means compiling to typescript is not supported (which is fine) or if there is even an issue providing typescript solid definition files for integration (??).

Thank you for your point of view.
Dino

**Comment by alfonsogarciacaro** [comment, marked as answer]
Hi @D1no! Sorry for the late reply. There's currently some support for Typescript compilation using the `--typescript` flag but it's still experimental and I think it still needs to work to be usable (I haven't personally worked in the feature). By default Fable doesn't output .d.ts type declarations. A while ago I wrote [a plugin to generate a .d.ts declaration](https://github.com/fable-compiler/Fable.Store/blob/992942a6fade6ae88ba498bf053ac3bfaa77e7fc/samples/App/src/TodoMVC/TodoMVC.fs#L53-L56), originally intended to interact with Svelte, but it can be easily adapted to more general uses. I can do it if you're open to try it and report any bugs you notice.

**Reply by D1no** [reply]
Thank you! Ok good to know. Alright — having a generated js bundle that doesn't ship with first class `d.ts` sounds a little too risky for me for now 🤗. I'll explore some other functional programming languages and see whats out there. Then I might get back to your kind offer. Thank you!
